bot/exts/filters/filtering.py

```diff
@@ -179,15 +191,21 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
-    def get_name_matches(self, name: str) -> List[re.Match]:
-        """Check bad words from passed string (name). Return list of matches."""
-        name = self.clean_input(name)
-        matches = []
+    def get_name_match(self, name: str) -> Optional[re.Match]:
+        """Check bad words from passed string (name). Return the first match found."""
+        normalised_name = unicodedata.normalize("NFKC", name)
+        cleaned_normalised_name = "".join(c for c in normalised_name if not unicodedata.combining(c))
```

Comment:
```suggestion
        cleaned_normalised_name = "".join([c for c in normalised_name if not unicodedata.combining(c)])
```
List comprehension for that Sonic the hedgehog speed?

Reply:
Isn't not building an intermediate list faster?

Reply:
Apparently not, just tested in repl, giving str.join a list is about 10-20% faster that giving it a generator

Reply:
List comprehensions are faster yes. Without the brackets you are passing an iterable (generator in this case) which Python needs to create, then individually advance and resize the list.

`list(x for x in y)` is not the same as `[x for x in y]`.

That kind of stuff adds up.

Reply:
https://github.com/python-discord/bot/pull/1892/commits/16f389efa239f5847b1553c461bf19725f1f33bd